src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import * as todoService from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { TempTodo } from '../../types/TempTodo';
+import { TodoContext } from '../../context/TodoContext';
+import { Errors } from '../../types/Errors';
+import { delay } from '../../utils/delay';
+import { focusInputField } from '../../utils/focusInputField';
+
+type Props = {
+  todo: Todo | TempTodo;
```

Comment:
```suggestion
  todo: Todo;
```
let's remove the TempTodo type from your app at all, it looks redundant